sdk-tests/src/test/java/io/dapr/it/spring/data/PostgreSQLDaprKeyValueTemplateIT.java

```diff
@@ -0,0 +1,232 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package io.dapr.it.spring.data;
+
+import io.dapr.client.DaprClient;
+import io.dapr.spring.data.DaprKeyValueTemplate;
+import io.dapr.testcontainers.Component;
+import io.dapr.testcontainers.DaprContainer;
+import io.dapr.testcontainers.DaprLogLevel;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.keyvalue.core.query.KeyValueQuery;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.containers.wait.strategy.WaitStrategy;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import java.util.*;
+
+import static io.dapr.it.spring.data.DaprSpringDataConstants.BINDING_NAME;
+import static io.dapr.it.spring.data.DaprSpringDataConstants.STATE_STORE_NAME;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * Integration tests for {@link PostgreSQLDaprKeyValueTemplateIT}.
+ */
+@SuppressWarnings("AbbreviationAsWordInName")
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = TestDaprSpringDataConfiguration.class)
+@Testcontainers
+@Tag("testcontainers")
+public class PostgreSQLDaprKeyValueTemplateIT {
+  private static final String CONNECTION_STRING =
+      "host=postgres user=postgres password=password port=5432 connect_timeout=10 database=dapr_db";
+  private static final Map<String, String> STATE_STORE_PROPERTIES = createStateStoreProperties();
+
+  private static final Map<String, String> BINDING_PROPERTIES = Collections.singletonMap("connectionString", CONNECTION_STRING);
+
+  private static final Network DAPR_NETWORK = Network.newNetwork();
+
+  @Container
+  private static final PostgreSQLContainer<?> POSTGRE_SQL_CONTAINER = new PostgreSQLContainer<>("postgres:16-alpine")
+      .withNetworkAliases("postgres")
+      .withDatabaseName("dapr_db")
+      .withUsername("postgres")
+      .withPassword("password")
+      .withExposedPorts(5432)
+      .withNetwork(DAPR_NETWORK);
+
+  private static final WaitStrategy DAPR_CONTAINER_WAIT_STRATEGY = Wait.forHttp("/v1.0/healthz")
```

Comment:
@artur-ciocanu maybe we can make this internal.. meaning that this can be the default strategy. If I remember correctly, depending the parameters that we set we might run into a loop of `daprd` waiting for the app port to be exposed and the test waiting for `daprd` to be ready. But as a user I don't want to be dealing with this.